CRM/Civicase/Hook/BuildForm/CaseTypeCategoryWebFormRedirect.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+use CRM_Civicase_Service_CaseCategorySetting as CaseCategorySetting;
+use CRM_Civicase_Hook_Helper_CaseTypeCategory as CaseTypeCategoryHelper;
+
+/**
+ * Fetches web form url if available for current case type category and redirects user to that web form.
+ */
+class CRM_Civicase_Hook_BuildForm_CaseTypeCategoryWebFormRedirect {
```

Comment:
Can we try use the PreProcess hook instead of the build form?
The reason is that the before the build form hook runs, a lot of business logic would have been done which is not necessary in case a redirect is needed.

Using a preprocess hook will help save a bit of time and running of un-necessary business logic if form will have to be redirected.

Reply:
sure will change it.